Library/ViewModels/ShippingRulesViewModel.swift

```diff
@@ -32,70 +33,73 @@ public protocol ShippingRulesViewModelType {
 public final class ShippingRulesViewModel: ShippingRulesViewModelType,
   ShippingRulesViewModelInputs, ShippingRulesViewModelOutputs {
   public init() {
-    let dataInitial = Signal.combineLatest(
+    let initialData = Signal.combineLatest(
       self.viewDidLoadProperty.signal,
       self.configDataProperty.signal
     )
     .map(second)
     .skipNil()
 
-    let selectedIndex = Signal.combineLatest(
-      dataInitial,
-      self.didSelectShippingRuleAtIndexProperty.signal.skipNil()
-    )
-    .map { data, newIndex in (data.1.firstIndex(of: data.2), newIndex) }
-    .filter { oldIndex, newIndex in oldIndex != newIndex }
-    .map(second)
+    let searchText = self.searchTextDidChangeProperty.signal
+      .ksr_debounce(.milliseconds(100), on: AppEnvironment.current.scheduler)
+      .skipNil()
 
-    let dataSelected = Signal.combineLatest(
-      dataInitial,
-      selectedIndex
-    )
-    .map { data, index in (data.0, data.1, index) }
-    .filter { _, shippingRules, index in index >= 0 && index < shippingRules.count }
-    .map { project, shippingRules, index in (project, shippingRules, shippingRules[index]) }
+    let selectedIndex = self.didSelectShippingRuleAtIndexProperty.signal
+      .skipNil()
 
-    let reloadDataWithShippingRulesInitial = dataInitial
-      .map(shippingRuleData(for:shippingRules:selectedShippingRule:))
-      .map { ($0, true) }
+    self.deselectCellAtIndex = selectedIndex
```

Comment:
It looks like in `ShippingRulesTableViewController` where we observe `deselectCellAtIndex` we don't actually use the index anymore and just deselect all cells. Can we rename this output to reflect that? Maybe  something like `let deselectCells: Signal<Void, Never>`?